### PR TITLE
Switch renderer from OpenGL to ANGLE

### PR DESCRIPTION
**Pull request by Wires77**
To be done before merging:

- [ ] Merge in master
- [x] Update CI script
- [x] Default ImGUI to off
- [x] Find out why debugging Lua doesn't let PoB launch
- [ ] For running on wine (Running in an Ubuntu VM) it looks like we will need to also build VCRUNTIME140D.dll and ucrtbased.dll in order for it to work.

Nice-to-haves for later PRs:
- [x] Update SG version (1 -> 2)
- [x] Update README.md
- [ ] Update PathOfBuilding-Launcher to support this update
- [x] Remove .sln and .vcxproj files
- [x] Update vcpkg hash
- [x] Remove vcpkg.txt and vcpkg-ports

### CODE REVIEW

File: vcpkg.txt
Comment (Wires77) [inline vcpkg.txt:1]: Do we still need this file now that we have vcpkg.json?

**Review by Wires77 — COMMENTED** [review]
We don't need the vcpkg-ports folder anymore, right?  Didn't they add a luajit version upstream we can use?

File: vcpkg.txt
Comment (zao) [reply vcpkg.txt:1]: The `vcpkg.txt` files are indeed superfluous now in manifest mode as we have it all declared in `vcpkg.json`.

**Comment by zao** [comment]
> We don't need the vcpkg-ports folder anymore, right? Didn't they add a luajit version upstream we can use?

The local ports tree is currently unused, we are currently using a LuaJIT port from vcpkg upstream as we define which baseline vcpkg commit and what overlay ports to use in `vcpkg-configuration.json`.
I didn't remove the tree as I was uncertain if upstream LuaJIT would be good enough.

File: engine/system/win/sys_console_unix.cpp [hunk 1]
```diff
@@ -0,0 +1,158 @@
+// SimpleGraphic Engine
+// (c) David Gowor, 2014
```
Comment (Wires77) [inline engine/system/win/sys_console_unix.cpp:2]: Technically this isn't their work ;)

File: engine/system/win/sys_opengl.cpp
Comment (Wires77) [inline engine/system/win/sys_opengl.cpp:1]: Wow, surprised that most of this isn't necessary with GLFW

File: engine/system/win/sys_video.cpp [hunk 2]
```diff
@@ -212,137 +205,218 @@ int sys_video_c::Apply(sys_vidSet_s* set)
 	vid.pos[1] = wrec.top;
 	wrec.right = wrec.left + cur.mode[0];
 	wrec.bottom = wrec.top + cur.mode[1];
-	AdjustWindowRectEx(&wrec, style, FALSE, exStyle);
+	// TODO(LV): Verify that stored coordinates are aligned right.
```
Comment (Wires77) [inline engine/system/win/sys_video.cpp:208]: Showstopping TODO or nice-to-have?

**Review by Wires77 — COMMENTED** [review]
I saw some support for other OSes in here, is the only thing unsupported opening URLs?

**Comment by Wires77** [comment]
So one big issue I'm finding is that if I try and attach a debugger on the Lua side via PyCharm, PoB doesn't open and also doesn't kill the process.  That invocation looks like this in Launch.lua:OnInit():
```lua
package.cpath = package.cpath .. ';C:/Users/trevo/AppData/Roaming/JetBrains/PyCharmCE2021.1/plugins/EmmyLua/debugger/emmy/windows/x86/?.dll'
local dbg = require('emmy_core')
dbg.tcpListen('localhost', 9966)
dbg.waitIDE()
```

I tried to use `GetCloudProvider` on Windows 11 using these reproduction steps: https://github.com/PathOfBuildingCommunity/PathOfBuilding/pull/5390#issuecomment-1367090234 Unfortunately, the output of the function showed a status of 1 for both a file that had been freed up, and one that was available...

For running on wine (Running in an Ubuntu VM) it looks like we will need to also build `VCRUNTIME140D.dll` and `ucrtbased.dll` in order for it to work.

Finally, we need to hide the ImGUI UI by default.

**Comment by Wires77** [comment]
> On the debugger - you seem to use the `x86` rather than the `x64` DLL. The docs need an update around this assuming it's the only problem. I could attach to the process (which had no window at the time) with PyCharm's Emmy plugin.
Doh, I just kept using my existing debug lines, forgetting they would have different DLLs.  Works fine now as well.

> The cloud state is more about the provider as a whole rather than individual files as it queries the state of the sync root that contains the path. I couldn't find any conclusive state to query for whether a file could be successfully obtained apart from attempting to actually read it. The cloud status mostly indicates if the service is running and as such has a chance to hydrate files, with 0 being disconnected, 1 being connected and values up in the `0x8000'0000`s suggest that a new file hasn't been ingested yet.
It's better than nothing, we can see what kind of things we can do with that information at least.

> We should never have a dependency on the Debug runtime in a shipping situation and custom is to not ever deploy it outside of a development machine as the license doesn't allow redistribution. To run a Debug build on a machine that doesn't have a dev environment you need to privately deploy those non-redist files - were you running a Debug build or has aspects of it crept into the release?

I found it easiest to pull the built binaries onto my VM via this PR: https://github.com/PathOfBuildingCommunity/PathOfBuilding/pull/6705  They were built with these commands:
```shell
cmake -A x64 -G "Visual Studio 17 2022" --toolchain .\vcpkg\scripts\buildsystems\vcpkg.cmake -B build-output -DCMAKE_INSTALL_PREFIX=runtime
cmake --build build-output --config RelWithDebInfo -t INSTALL
```

**Pull request merged**